controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -520,6 +526,11 @@ func (r *BareMetalHostReconciler) actionDeleting(prov provisioner.Provisioner, i
 		"timestamp", info.host.DeletionTimestamp,
 	)
 
+	// Deleting DataImage if it exists for the BMH
+	if err := r.removeFinalizerDataImage(info); err != nil {
+		info.log.Error(err, "Failed to delete DataImage")
```

Comment:
Why do we ignore the error here? I'm fine with ignoring HTTP 404 (in case somebody did a nasty force-deletion), but it seems that ignoring any errors may result with orphaned DataImage.